simulator/mips/mips_instr.cpp

```diff
@@ -216,21 +226,57 @@ FuncInstr::FuncInstr( uint32 bytes, Addr PC,
 void FuncInstr::initFormat()
 {
     bool is_R = ( instr.asR.opcode == 0x0);
+    bool is_RI = ( instr.asR.opcode == 0x1);
     uint8 ident = is_R ? instr.asR.funct : instr.asR.opcode;
 
-    for ( const auto& entry : isaTable)
+    if ( is_R)
+    {    
+        const auto& entry = isaMapR.find( ident);
```

Comment:
Could we avoid code duplication here? If we decide to add a new field to ISAEntry, we have to add it in 3 places. It is easy to forget one of them and spend many hours to find a error.

I suggest to use the following code:

```c++
    bool valid = false;
    auto it = isaMapRI.cbegin(); // I am too lazy to write the type explicitly
    switch ( instr.asR.opcode) {
    case 0x0: // R instruction
           it = isaMapR.find( instr.asR.funct);
           valid = ( it != isaMapR.end());
           break;
// ....
```

Reply:
Please do that change and wait for testing results. If everything is OK, I'll merge the pull request.